File: Shell_Sort/Shell_Sort.php
<?php

function shellSort(&$arr){

	for ($gap=floor(count($arr)/2); $gap > 0; $gap= floor($gap/2)) { 
		for ($i=$gap; $i < count($arr); $i++) { 
			
			$temp = $arr[$i];
			$j = $i;
			for (; $j >= $gap && $arr[$j - $gap] > $temp; $j -= $gap) { 
				$arr[$j] = $arr[$j - $gap];
			}
			$arr[$j] = $temp;
		}
	}
}

function printArray($arr){

	$output = "[";
	for ($i=0; $i < count($arr); $i++) { 
		$output .=" ".$arr[$i]." ";
		if($i < count($arr) - 1)
			$output .=",";
	}
	$output .="]";
	echo $output;
	return;
}

$n = (int)readline();
$arr = array();

for ($i=0; $i < $n; $i++) { 
	$arr[$i] = (int)readline();
}

shellSort($arr);
printArray($arr);


?>

File: Shell_Sort/Shell_Sort.cpp
#include<iostream>
using namespace std;

void shellSort(int arr[],int n){

	int temp,i,j;
	for (int gap = n/2 ; gap > 0; gap = gap/2){
		for (i = gap; i < n; ++i){
			temp = arr[i];
			for (j = i; j>=gap && arr[j-gap] > temp; j = j - gap){
				arr[j] = arr[j-gap];
			}
			arr[j] = temp;
		}
	}

	return;
}

	void printArray(int arr[],int n){

	cout<<"[";
	for (int i = 0; i < n; ++i){
		cout<<" "<<arr[i]<<" ";
		if(i<n-1)
			cout<<",";
	}
	cout<<"]";
	return;
}

int main(int argc, char const *argv[]){
	int n;
	cin>>n;
	int arr[n];

	for (int i = 0; i < n; ++i){
		cin>>arr[i];
	}

	shellSort(arr,n);
	printArray(arr,n);
	return 0;
}